### PR TITLE
Do not report confusing error

### PR DESCRIPTION
**Pull request by alfbr**
Removed a confusing error message from LOG/PRT. @tskille please see if this fixes your issue.

### CODE REVIEW

**Comment by tskille** [comment]
Thanks @abir, confusing error messages gone.

**Comment by alfbr** [comment]
Good, thanks for testing!

**Comment by alfbr** [comment]
jenkins build this please

**Comment by alfbr** [comment]
This is ready for review from my end.

**Comment by GitPaean** [comment]
Hi, with this way, the error message will not show up anywhere, include .DBG file.  I think we should have this information in the .DBG file to help the developers. 

Maybe we should have a `OpmLog::debug()` before we `OPM_THROW_NOLOG` if we do not want develop new facility? Any suggestions @bska @joakim-hove ?

**Comment by bska** [comment]
> Maybe we should have a `OpmLog::debug()` before we `OPM_THROW_NOLOG` if we do not want develop new facility?

Yes, that'd be the least effort alternative here.  That way we at least get a trace of the problem in one of the
```
CASE.N.DBG
```
files (or just `CASE.DBG` if rank zero ends up in this situation).

**Comment by bska** [comment]
> That way we at least get a trace of the problem in one of the

We still have the problem that we don't terminate if one process/rank gets stuck after throwing an exception though

**Comment by GitPaean** [comment]
> Yes, that'd be the least effort alternative here. That way we at least get a trace of the problem in one of the

Thanks for the input. 

> We still have the problem that we don't terminate if one process/rank gets stuck after throwing an exception though

Yes. It will be more involved and out of the scope of this PR.

**Comment by GitPaean** [comment]
> We still have the problem that we don't terminate if one process/rank gets stuck after throwing an exception though

I can try to do it in a separate PR based on DeferredLogger.

File: opm/simulators/wells/MSWellHelpers.hpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ namespace mswellhelpers
         for (size_t i_block = 0; i_block < y.size(); ++i_block) {
             for (size_t i_elem = 0; i_elem < y[i_block].size(); ++i_elem) {
                 if (std::isinf(y[i_block][i_elem]) || std::isnan(y[i_block][i_elem]) ) {
-                    OPM_THROW(Opm::NumericalIssue, "nan or inf value found after UMFPack solve due to singular matrix");
+                    OPM_THROW_NOLOG(Opm::NumericalIssue, "nan or inf value found after UMFPack solve due to singular matrix");
```
Comment (GitPaean) [inline opm/simulators/wells/MSWellHelpers.hpp:67]: Referring to the discussion in the comments, the following is suggested here, 

```c++
        const std::string msg{"nan or inf value found after UMFPack solve due to singular matrix"};
        OpmLog::debug(msg);
        OPM_THROW_NOLOG(Opm::NumericalIssue, msg);
```

You need to add the following header file
```
#include <opm/common/OpmLog/OpmLog.hpp>
#include <string>
```

File: opm/simulators/wells/MultisegmentWell_impl.hpp [hunk 2]
```diff
@@ -3185,7 +3185,7 @@ namespace Opm
 
             const EvalWell d = 1.0 - rs * rv;
             if (d <= 0.0 || d > 1.0) {
-                OPM_THROW(Opm::NumericalIssue, "Problematic d value " << d << " obtained for well " << name()
```
Comment (GitPaean) [inline opm/simulators/wells/MultisegmentWell_impl.hpp:3188]: Similar to the previous comment, the following code is suggested, 
```c++
                std::ostringstream sstr;
                sstr << "Problematic d value " << d << " obtained for well " << name()
                     << " during conversion to surface volume with rs " << rs
                     << ", rv " << rv << " and pressure " << seg_pressure
                     << " obtaining d " << d;
                OpmLog::debug(sstr.str());
                OPM_THROW_NOLOG(Opm::NumericalIssue, sstr.str());
```

**Review by GitPaean — CHANGES_REQUESTED** [review]
We need to keep some debugging output in the .DBG file. So `OpmLog::debug()` is suggested to call before the `OPM_THROW_NOLOG` call.

**Comment by alfbr** [comment]
jenkins build this please

**Comment by alfbr** [comment]
Review comments addressed and jenkins seems happy.

**Comment by GitPaean** [comment]
> Review comments addressed and jenkins seems happy.

Thanks for the efforts. I am merging it now.

**Pull request merged**